File: datasets/C++/Deepseek/DataStructures/medium1.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> matchingStrings(vector<string> stringList, vector<string> queries) {
    unordered_map<string, int> countMap;
    for (const string &s : stringList) {
        countMap[s]++;
    }
    vector<int> result;
    for (const string &q : queries) {
        result.push_back(countMap[q]);
    }
    return result;
}

int main() {
    // Taking input manually from the user
    int n;
    cout << "Enter the number of strings in the list: ";
    cin >> n;
    vector<string> stringList(n);
    for (int i = 0; i < n; ++i) {
        cout << "Enter string " << i+1 << ": ";
        cin >> stringList[i];
    }
    int q;
    cout << "Enter the number of queries: ";
    cin >> q;
    vector<string> queries(q);
    for (int i = 0; i < q; ++i) {
        cout << "Enter query " << i+1 << ": ";
        cin >> queries[i];
    }
    vector<int> results = matchingStrings(stringList, queries);
    cout << "Results: ";
    for (int res : results) {
        cout << res << " ";
    }
    cout << endl;
    return 0;
}

File: datasets/Python/Deepseek/DataStructures/medium1.py
def matchingStrings(stringList, queries):
    result = []
    count_dict = {}
    for s in stringList:
        if s in count_dict:
            count_dict[s] += 1
        else:
            count_dict[s] = 1
    for q in queries:
        result.append(count_dict.get(q, 0))
    return result

if __name__ == '__main__':
    # Taking input manually from the user
    n = int(input("Enter the number of strings in the list: "))
    stringList = []
    for i in range(n):
        s = input(f"Enter string {i+1}: ")
        stringList.append(s)
    q = int(input("Enter the number of queries: "))
    queries = []
    for i in range(q):
        query = input(f"Enter query {i+1}: ")
        queries.append(query)
    results = matchingStrings(stringList, queries)
    print("Results:", results)
